features/all-positions/AllLiquidations.tsx

```diff
@@ -0,0 +1,330 @@
+
+  // Sorting
+  const [sortDirection, setSortDirection] = useState<boolean>(true);
+  const [sortedColumn, setSortedColumn] = useState<number>(
+    SORT_FIELD.TOKENS_LIQUIDATED
+  );
+
+  // Set max page depending on # of liqs
+  useEffect(() => {
+    setMaxPage(1);
+    setPage(1);
+
+    if (liquidations) {
+      let extraPages = 1;
+      if (Object.keys(liquidations).length % ITEMS_PER_PAGE === 0) {
+        extraPages = 0;
+      }
+      setMaxPage(
+        Math.floor(Object.keys(liquidations).length / ITEMS_PER_PAGE) +
+          extraPages
+      );
+      // This will set maxPage to 0 if there are no liqs.
+    }
+  }, [liquidations]);
+
+  if (
+    tokenSymbol !== null &&
+    collSymbol !== null &&
+    priceId !== null &&
+    liquidations &&
+    liquidationLiveness !== null &&
+    Object.keys(liquidations).length > 0
+  ) {
+    const prettyBalance = (x: number) => {
+      const x_string = x.toFixed(4);
+      return utils.commify(x_string);
+    };
+
+    const prettyDate = (x_secs: number) => {
+      return new Date(x_secs * 1000).toLocaleString("en-GB", {
+        timeZone: "UTC",
+      });
+    };
+
+    const prettyAddress = (x: string) => {
+      return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+    };
+
+    const invertDisputablePrice = isPricefeedInvertedFromTokenSymbol(
+      tokenSymbol
+    );
+    const getDisputablePrice = (x: string) => {
+      return invertDisputablePrice && parseFloat(x) !== 0
+        ? 1 / Number(x)
+        : Number(x);
+    };
+
+    const translateLiquidationStatus = (
+      liquidationTimestamp: number,
+      liquidationStatus: string
+    ) => {
+      const liquidationTimeRemaining =
+        liquidationTimestamp +
+        liquidationLiveness.toNumber() -
+        Math.floor(Date.now() / 1000);
+      if (liquidationTimeRemaining > 0 && liquidationStatus === "PreDispute") {
+        return "Liquidation Pending";
+      } else if (
+        liquidationTimeRemaining <= 0 &&
+        liquidationStatus === "PreDispute"
+      ) {
+        return "Liquidation Succeeded";
+      } else if (liquidationStatus === "PendingDispute") {
+        return "Dispute Pending";
+      } else if (liquidationStatus === "DisputeSucceeded") {
+        return "Dispute Succeeded";
+      } else if (liquidationStatus === "DisputeFailed") {
+        return "Dispute Failed";
+      } else {
+        return "Unknown";
+      }
+    };
+
+    // First filters out liq. data missing field values,
+    // then sorts the positions according to selected sort column,
+    // and finally slices the array based on pagination selection.
+    const reformattedLiquidationData = Object.keys(liquidations)
+      .filter((sponsor: string) => {
+        return (
+          liquidations[sponsor]?.maxDisputablePrice &&
+          liquidations[sponsor]?.tokensLiquidated &&
+          liquidations[sponsor]?.lockedCollateral &&
+          liquidations[sponsor]?.liquidationTimestamp
```

Comment:
Quick question: it's desired to filter these out if any of them are 0 in addition to being null, right?

Reply:
Yeah I suppose thats right. We can filter them out in the `EMPSponsors` container